src/main/java/org/isf/medicalinventory/gui/InventoryWardEdit.java

```diff
@@ -336,88 +466,257 @@ private JPanel getPanelFooter() {
             String user = UserBrowsingManager.getCurrentUser();
-            int checkResults = 0;
-            if (inventoryRowSearchList == null || inventoryRowSearchList.size() < 1) {
-                MessageDialog.error(null, "angal.inventory.noproduct.msg");
-                return;
-            }
-            LocalDateTime now = LocalDateTime.now();
-            if (dateInventory.isAfter(now)) {
-                MessageDialog.error(null, "angal.inventory.notdateinfuture.msg");
+            if (inventoryRowSearchList == null || inventoryRowSearchList.isEmpty()) {
+                MessageDialog.error(null, "angal.inventory.cannotsaveinventorywithoutproducts.msg");
                 return;
             }
-
-            if ((inventory == null) && (mode.equals("new"))) {
-                String reference = referenceTextField.getText().trim();
-                if (reference.equals("")) {
-                    MessageDialog.error(null, "angal.inventory.mustenterareference.msg");
-                    return;
-                }
-                if (medicalInventoryManager.referenceExists(reference)) {
-                    MessageDialog.error(null, "angal.inventory.referencealreadyused.msg");
-                    return;
+            try {
+                if (!inventoryRowsToDelete.isEmpty()) {
+                    medicalInventoryRowManager.deleteMedicalInventoryRows(inventoryRowsToDelete);
                 }
-                inventory = new MedicalInventory();
-                inventory.setInventoryReference(reference);
-                inventory.setInventoryDate(dateInventory);
-                inventory.setStatus(State);
-                inventory.setUser(user);
-                inventory.setInventoryType(InventoryType.main.toString());
-                MedicalInventory meInventory;
-                try {
-                    meInventory = medicalInventoryManager.newMedicalInventory(inventory);
-                    if (meInventory != null) {
-                        MedicalInventoryRow currentInventoryRow;
-                        for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator
-                                .hasNext();) {
-                            MedicalInventoryRow medicalInventoryRow = (MedicalInventoryRow) iterator.next();
-                            medicalInventoryRow.setInventory(meInventory);
-                            Lot lot = medicalInventoryRow.getLot();
-                            if (lot != null && lot.getCode().equals("")) {
-                                medicalInventoryRow.setLot(null);
+	            chargeType = (MovementType) chargeComboBox.getSelectedItem();
+	            dischargeType = (MovementType) dischargeComboBox.getSelectedItem();
+	            supplier = (Supplier) supplierComboBox.getSelectedItem();
+	            destination = (Ward) destinationComboBox.getSelectedItem();
+                if ((inventory == null) && (mode.equals("new"))) {
```

Comment:
```suggestion
                if (inventory == null && mode.equals("new")) {
```
Unnecessary parentheses; there seems to be about 20 other such cases (which I will not enumerate).
Parentheses are considered unnecessary if the evaluation order of an expression remains unchanged after the removal of the parentheses.